VideoAPI/VideoApi.DAL/Queries/GetInstantMessagesForConferenceQuery.cs

```diff
@@ -37,6 +37,13 @@ public async Task<List<InstantMessage>> Handle(GetInstantMessagesForConferenceQu
                 .OrderByDescending(x => x.TimeStamp)
                 .ToListAsync();
 
+            if(query.ParticipantName != null)
+            {
```

Comment:
this query could be all kept to sql

If you avoid calling `ToListAsync` until you return then the query will all execute as a sql query

Reply:
update to return list after filtering.